R/combined.R

```diff
@@ -286,6 +297,55 @@ combined_aggregate_data <- function(data) {
 }
 
 
+combined_aggregate_severity <- function(severity, samples) {
+
+  # We first need to sum admitted_inc and diagnoses_inc trajectories to get
+  # all_admissions_inc, which we'll use as weight to aggregate hospital severity
+  for (i in names(samples)) {
+    admitted_inc <- samples[[i]]$trajectories$state["admitted_inc", , ]
+    diagnoses_inc <- samples[[i]]$trajectories$state["diagnoses_inc", , ]
+    all_admissions <- array(admitted_inc + diagnoses_inc, dim = c(1, 10, 744))
+    rownames(all_admissions) <- "all_admissions_inc"
+    samples[[i]]$trajectories$state <-
+      abind_quiet(samples[[i]]$trajectories$state, all_admissions, along = 1)
+  }
+
+  # Some unfortunate double running of the same function outputting a similar
+  # object but with different weighting
+  admissions <- combined_aggregate_severity_1(severity, samples,
+                                              weight = "all_admissions_inc")
+
+  infections <- combined_aggregate_severity_1(severity, samples,
+                                              weight = "infections_inc")
+
+  # Vector of severity trajectories weighted by admissions
+  # Note this is only one for now, but will grow going forward
+  admission_weighted <- "ihr"
+  severity$england <- infections$england
+  for (i in admission_weighted) {
+    severity$england[[i]] <- admissions$england[[i]]
+  }
+
+  severity
+}
+
+
+combined_aggregate_severity_1 <- function(x, samples, weight) {
+  england <- sircovid::regions("england")
+  nations <- sircovid::regions("nations")
+
+  if (all(england %in% names(x))) {
+    x$england <- sircovid::combine_rt(x[england], samples[england],
+                                       rank = FALSE, weight = weight)
+  }
+  if (all(nations %in% names(x))) {
+    stop("We cannot currently run severity analysis at DVN/UK level!")
```

Comment:
Why do we need this?

Reply:
changed

Reply:
Why not aggregate to UK level though - I don't see why we can't here